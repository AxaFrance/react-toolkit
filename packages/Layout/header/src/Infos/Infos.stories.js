import React from 'react';
import { storiesOf } from '@storybook/react';
import Infos from './Infos';
import readme from '../../README.md';

const infos = [
  {
    word: 'Client :',
    definition: '0123456789 - NOM',
  },
  {
    word: 'Portefeuille :',
    definition: '000123456789',
  },
  {
    word: 'Status :',
    definition: 'Affaire nouvelle',
  },
];

const InfosStory = () => <Infos infos={infos} />;

const stories = storiesOf('Layout/Header/Infos', module);

stories.addParameters({
  readme: {
    sidebar: readme,
  },
  options: {},
});

stories.add('Default', InfosStory);
