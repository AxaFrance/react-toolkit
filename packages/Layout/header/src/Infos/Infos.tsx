import React, { Fragment } from 'react';
import { ClassManager, InputManager } from '@axa-fr/react-toolkit-core';

const defaultClassName = 'af-contrat';
type Props = {
  infos: { id?: string; word: string; definition: string }[];
  className?: string;
  classModifier?: string;
};

const Infos = ({ infos, className, classModifier }: Props) => {
  const componentClassName = ClassManager.getComponentClassName(
    className,
    classModifier,
    defaultClassName
  );

  return (
    <div className={componentClassName}>
      <i className="glyphicon glyphicon-info-sign" />
      <dl className={`${defaultClassName}__list`}>
        {infos.map((info) => (
          <Fragment key={info.id}>
            <dt
              key={`word_${InputManager.getInputId(info.id)}`}
              className={`${defaultClassName}__word`}>
              {info.word}
            </dt>
            <dd
              key={`def_${InputManager.getInputId(info.id)}`}
              className={`${defaultClassName}__def`}>
              {info.definition}
            </dd>
          </Fragment>
        ))}
      </dl>
    </div>
  );
};

export default Infos;
