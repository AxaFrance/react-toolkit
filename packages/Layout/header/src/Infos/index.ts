export { default as Infos } from './Infos';
