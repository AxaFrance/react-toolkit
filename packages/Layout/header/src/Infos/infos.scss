@import '@axa-fr/react-toolkit-core/src/common/scss/core.scss';
.af-contrat {
  position: relative;
  display: flex;
  justify-content: space-between;
  align-items: center;
  border: 1px solid $gray-lighter;
  border-radius: 25px;
  padding: 0.4rem 0.3rem 0.4rem 3rem;
  font-size: 0.9em;
  order: 3;

  .glyphicon-info-sign {
    position: absolute;
    color: $color-gray-contract;
    font-size: 1.25rem;
    top: 50%;
    left: 0.7rem;
    transform: translate(0, -50%);
  }

  &__list {
    margin: 0;
  }

  &__word {
    display: inline-block;
    margin-right: 0.2rem;
  }

  &__def {
    display: inline-block;
    margin: 0 0.5rem 0 0;
  }
}

@include media-breakpoint-up(md) {
  .af-contrat {
    order: 2;
    margin: 0;
  }
}
