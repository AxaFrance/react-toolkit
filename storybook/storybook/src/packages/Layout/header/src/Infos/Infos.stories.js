import React from 'react';
import { storiesOf } from '@storybook/react';
import { Infos } from '@axa-fr/react-toolkit-layout-header';
import readme from '@axa-fr/react-toolkit-layout-header/dist/README.md';

const infos = [
  {
    word: 'Client :',
    definition: '0123456789 - NOM',
  },
  {
    word: 'Portefeuille :',
    definition: '000123456789',
  },
  {
    word: 'Status :',
    definition: 'Affaire nouvelle',
  },
];

const InfosStory = () => <Infos infos={infos} />;

const stories = storiesOf('Layout/Header/Infos', module);

stories.addParameters({
  readme: {
    sidebar: readme,
  },
  options: {},
});

stories.add('Default', InfosStory);
