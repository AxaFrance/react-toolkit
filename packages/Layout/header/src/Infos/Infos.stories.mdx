import { Meta, Story, Canvas } from '@storybook/addon-docs';
import { ArgsTable } from '@storybook/addon-docs';
import Infos from './Infos';

<Meta title="Structure/Header/Infos" component={Infos} argTypes={{}} />

# Infos

export const Template = (args) => (
  <Infos {...args} />
);

The header package exports an `Infos` in order to display a list of descriptions.  
To do so, have an `infos` property which is an array of object that will be shown one after another into the header. That array has a particular type and contains the following properties :
- an `id`, which is optionnal
- a `word`, which will specify the term
- a `definition`, which will specify the element description

Example : 

```javascript
const infos = [
  {
    word: 'Client :',
    definition: '0123456789 - NOM',
  },
  {
    word: 'Portefeuille :',
    definition: '000123456789',
  },
  {
    word: 'Status :',
    definition: (
      <span style={{ textDecoration: 'underline' }}>Affaire nouvelle</span>
    ),
  },
];
```

Don't forget to load the css file of the component.

```javascript
import '@axa-fr/react-toolkit-layout-header/dist/Infos/af-infos.css';
```

export const infos = [
  {
    word: 'Client :',
    definition: '0123456789 - NOM',
  },
  {
    word: 'Portefeuille :',
    definition: '000123456789',
  },
  {
    word: 'Status :',
    definition: (
      <span style={{ textDecoration: 'underline' }}>Affaire nouvelle</span>
    ),
  },
];

<Canvas>
  <Story
    name="Default"
    args={{
      infos
    }}>
    {Template.bind({})}
  </Story>
</Canvas>

<ArgsTable story="Default" />
