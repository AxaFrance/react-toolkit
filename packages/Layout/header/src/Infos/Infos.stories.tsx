import React from 'react';
import { Meta } from '@storybook/react';
import Infos from './Infos';
import readme from '../../README.md';

export default {
  title: 'Layout/Header/Infos',
  component: Infos,
  parameters: {
    readme: {
      sidebar: readme,
    },
    options: {},
  },
} as Meta;

const infos = [
  {
    word: 'Client :',
    definition: '0123456789 - NOM',
  },
  {
    word: 'Portefeuille :',
    definition: '000123456789',
  },
  {
    word: 'Status :',
    definition: 'Affaire nouvelle',
  },
];

export const Default = () => <Infos infos={infos} />;
