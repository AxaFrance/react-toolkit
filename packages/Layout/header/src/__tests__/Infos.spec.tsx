import React from 'react';
import { render } from '@testing-library/react';
import Infos from '../Infos/Infos';

describe('<Infos>', () => {
  it('renders Infos correctly', () => {
    const { asFragment } = render(
      <Infos infos={[{ word: 'word', definition: 'definition', id: '1' }]} />
    );
    expect(asFragment).toMatchSnapshot();
  });
});
