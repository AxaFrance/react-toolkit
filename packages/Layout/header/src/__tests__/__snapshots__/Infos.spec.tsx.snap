// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Infos> renders Infos correctly 1`] = `[Function]`;
